template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreCloudStorageRoleConfigProvider.java

```diff
@@ -0,0 +1,70 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.hive;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRoleConfigConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import com.sequenceiq.cloudbreak.template.views.HostgroupView;
+
+@Component
+public class HiveMetastoreCloudStorageRoleConfigProvider extends AbstractRoleConfigConfigProvider {
+
+    private static final String HMS_CONFIG_SAFETY_VALVE = "hive_metastore_config_safety_valve";
+
+    private static final String HMS_METASTORE_DIR = "hive.metastore.warehouse.dir";
+
+    private static final String HMS_METASTORE_EXTERNAL_DIR = "hive.metastore.warehouse.external.dir";
+
+    @Override
+    protected List<ApiClusterTemplateConfig> getRoleConfig(String roleType, HostgroupView hostGroupView, TemplatePreparationObject source) {
+        switch (roleType) {
+            case HiveRoles.HIVEMETASTORE:
+                List roleConfigs = new ArrayList(1);
+                getCloudStorageProperties(source)
+                        .filter(s -> !s.isEmpty())
+                        .ifPresent(storage -> roleConfigs.add(config(HMS_CONFIG_SAFETY_VALVE, storage)));
```

Comment:
My CM understanding is rather limited, but is it OK to set the same config key (safety valve in this case) multiple times for a given role/service? I mean, will CM indeed merge them into a single XML fragment, or just pick the one appearing last in the API request?

Reply:
Okay, never mind. Just see that `getCloudStorageProperties()` returns the whole thing as a single string.

Reply:
The HMS Safety valve can have multiple properties.